src/lambda_codebase/initial_commit/bootstrap_repository/adf-build/shared/helpers/terraform/adf_terraform.sh

```diff
@@ -0,0 +1,175 @@
+    # if TARGET_ACCOUNTS and TARGET_OUS are not defined apply to all accounts
+    if [[ -z "$TARGET_ACCOUNTS" ]] && [[ -z "$TARGET_OUS" ]]
+    then
+        echo "Apply to all accounts" 
+        for ACCOUNT_ID in $(jq '.[].AccountId' "$CURRENT"/accounts.json | sed 's/"//g' ) 
+        do
+            export TF_VAR_TARGET_ACCOUNT_ID=$ACCOUNT_ID
+            echo "Running terraform $TF_STAGE on account $ACCOUNT_ID and region $REGION"
+            if [[ "$TF_STAGE" = "plan" ]]
+            then
+                tfinit
+                if [[ $? -ne 0 ]]
+                then
+                    exit 1
+                fi
+                tfplan
+                if [[ $? -ne 0 ]]
+                then
+                    exit 1
+                fi
+            elif [[ "$TF_STAGE" = "apply" ]]
+            then
+                tfinit
+                if [[ $? -ne 0 ]]
+                then
+                    exit 1
+                fi
+                tfplan
+                if [[ $? -ne 0 ]]
+                then
+                    exit 1
+                fi
+                tfapply
+                if [[ $? -ne 0 ]]
+                then
+                    exit 1
+                fi
+            else 
+                echo "Invalid terraform command"
+            fi
+        done
+    fi
+
+    if ! [[ -z "$TARGET_ACCOUNTS" ]]
+    then
+        # apply only on a subset of accounts (TARGET_ACCOUNTS)
+        echo "List of target account: $TARGET_ACCOUNTS"
+        for ACCOUNT_ID in $(echo $TARGET_ACCOUNTS | sed "s/,/ /g")
+        do  
+            export TF_VAR_TARGET_ACCOUNT_ID=$ACCOUNT_ID
+            echo "Running terraform $TF_STAGE on account $ACCOUNT_ID and region $REGION"
+            if [[ "$TF_STAGE" = "plan" ]]
+            then
+                tfinit
+                if [[ $? -ne 0 ]]
+                then
+                    exit 1
+                fi
+                tfplan
+                if [[ $? -ne 0 ]]
+                then
+                    exit 1
+                fi
+            elif [[ "$TF_STAGE" = "apply" ]]
+            then
+                tfinit
+                if [[ $? -ne 0 ]]
+                then
+                    exit 1
+                fi
+                tfplan
+                if [[ $? -ne 0 ]]
+                then
+                    exit 1
+                fi
+                tfapply
+                if [[ $? -ne 0 ]]
+                then
+                    exit 1
+                fi
+            else 
+                echo "Invalid terraform command"
+            fi
+        done
+    fi
+
+    if ! [[ -z "$TARGET_OUS" ]]
+    then
+        echo "List target OUs: $TARGET_OUS" 
+        for ACCOUNT_ID in $(jq '.[].AccountId' "$CURRENT"/accounts_from_ous.json | sed 's/"//g' ) 
+        do
+            export TF_VAR_TARGET_ACCOUNT_ID=$ACCOUNT_ID
+            echo "Running terraform $TF_STAGE on account $ACCOUNT_ID and region $REGION"
+            if [[ "$TF_STAGE" = "plan" ]]
```

Comment:
Same here, about code repetition. Please use functions instead.

Reply:
Added the function tfrun to avoid code repetition